ide/projectui/src/org/netbeans/modules/project/ui/ProjectUtilities.java

```diff
@@ -135,28 +130,25 @@ private void doClose(Project[] projects, boolean notifyUI, Wrapper wr) {
                 if (dobj != null) {
                     FileObject fobj = dobj.getPrimaryFile();
                     Project owner = ProjectConvertors.getNonConvertorOwner(fobj);
-                    ERR.log(Level.FINER, "Found {0} owned by {1} in {2} of {3}", new Object[] {fobj, owner, tc.getName(), tc.getClass()});
+                    LOG.log(Level.FINER, "Found {0} owned by {1} in {2} of {3}", new Object[] {fobj, owner, tc.getName(), tc.getClass()});
 
-                    if (listOfProjects.contains(owner)) {
+                    if (owner != null && listOfProjects.contains(owner)) {
                         if (notifyUI) {
                             openFiles.add(dobj);
                             tc2close.add(tc);
                         } else if (!dobj.isModified()) {
                             // when not called from UI, only include TCs that arenot modified
                             tc2close.add(tc);
                         }
-                        //#235897 split a single line to detect NPE better
-                        final Set<String> pwnr = wr.urls4project.get(owner);
-                        assert pwnr != null : "Owner project for file:" + fobj + " prj:" + owner;
-                        final FileObject pf = fobj;
-                        assert pf != null;
-                        URL u = pf.toURL();
-                        assert u != null;
-                        String uex = u.toExternalForm();                        
-                        pwnr.add(uex);
+                        Set<String> files = project2FilesMap.get(owner);
+                        if (files != null) {
+                            files.add(fobj.toURL().toExternalForm());
+                        } else {
+                            LOG.log(Level.WARNING, "project accociation lost, project ({1}) might lose an opened file ({2}) on reopen", new Object[] {owner, fobj});
```

Comment:
How can this case happen? If files == null it means owner was never added to project2FilesMap. But all projects in listOfProjects were added to project2FilesMap before, and we are currently in an if condition that runs only if listOfProjects.contains(owner).

Reply:
broken equals/hashcode contracts.

After the added workaround which unboxes the project this shouldn't happen anymore. This is just a safety measure which prints something instead of throwing NPEs in case some edge case got overlooked.

Reply:
concrete example: the list of open projects might contain the `FeatureNonProject` wrapper, `owner` however is the real project -> contains/remove etc won't work and the code breaks